tezos/messages/src/p2p/encoding/operations_for_blocks.rs

```diff
@@ -7,25 +7,27 @@ use std::sync::Arc;
 use bytes::Buf;
 
 use getset::{CopyGetters, Getters};
-use serde::{Deserialize, Serialize, Serializer};
+use serde::{Deserialize, Serialize};
 
 use crypto::hash::{BlockHash, Hash, HashType};
 use tezos_encoding::binary_reader::BinaryReaderError;
 use tezos_encoding::encoding::{CustomCodec, Encoding, Field, HasEncoding};
 use tezos_encoding::ser::Error;
-use tezos_encoding::types::{RecursiveDataSize, Value};
+use tezos_encoding::types::Value;
 use tezos_encoding::{has_encoding, safe};
 
 use crate::cached_data;
 use crate::p2p::binary_message::cache::BinaryDataCache;
 use crate::p2p::encoding::operation::Operation;
 use tezos_encoding::json_writer::JsonWriter;
 
-/// Maximal depth for path to be decoded/encoded.
-/// Serde serialization is still recursive and can lead to stack overflow.
-/// This value can be handled by serde and still it is big enough.
-/// to handle 2 ** 512 elements tree.
-pub const PATH_MAX_DEPTH: RecursiveDataSize = 512;
+/// Maximal lenght for operation path to be decoded/encoded.
+/// This is in sync with Tezos limit for number of operations,
+/// https://gitlab.com/simplestaking/tezos/-/blob/master/src/lib_shell/distributed_db_message.ml#L65
+/// so path for a tree containing no more than 8 items
+/// should not exceed 3.
+/// TODO: Implement mechanism for updating this, when Tezos implements this.
+pub const PATH_MAX_SIZE: Option<usize> = Some(3);
```

Comment:
@akoptelov 
please, update this comments and name of constants more clearly, like we talked on slack

Reply:
@akoptelov 
and then rebase to develop and merge please